Exercise_1.java

```diff
@@ -1,44 +1,51 @@
-class Stack { 
-    static final int MAX = 1000; 
-    int top; 
+class Stack {
+    static final int MAX = 1000;
+    int top;
     int a[] = new int[MAX]; // Maximum size of Stack 
-  
-    boolean isEmpty() 
-    { 
-        //Write your code here 
-    } 
 
-    Stack() 
-    { 
-        //Initialize your constructor 
-    } 
-  
-    boolean push(int x) 
-    { 
+    Stack() {
+        top = 0;
+    }
+
+    boolean isEmpty() {
+        return top == 0;
+    }
+
+    boolean push(int x) {
         //Check for stack Overflow
-        //Write your code here
-    } 
-  
-    int pop() 
-    { 
+        if (top == MAX) {
+            System.out.println("Stack Overflow");
+            return false;
+        }
+
+        a[++top] = x;
+        return true;
+    }
+
+    int pop() {
         //If empty return 0 and print " Stack Underflow"
-        //Write your code here
-    } 
-  
-    int peek() 
-    { 
-        //Write your code here
-    } 
-} 
-  
+        if (top < 0) {

```

Comment:
The if statement will not guard against operating on an empty stack, top is initialized to 0 and will never be less than 0.

Reply:
The top pointer can go below zero as we keep popping elements. Doesn't matter what it was initialized to when creating the stack.